tls/server.go

```diff
@@ -0,0 +1,94 @@
+package tls
+
+import (
+	"crypto/tls"
+	"net"
+	"net/http"
+	"time"
+)
+
+// NetServer will serve a response on a socket with the provided server and listener,
+// and might do some other functions like supporting TLS if configured.
+type NetServer interface {
+	// Serve will serve the http.Server at the net.Listener address,
+	// but will perform optional other actions to support requirements,
+	// such as setting up TLSConfig to support TLS.
+	Serve(server *http.Server, listener net.Listener)
+}
+
+// TLSNetServer is a NetServer that adds TLS support to the provided http.Server and net.Listener.
+type TLSNetServer struct {
+	options *Options
+}
+
+var _ NetServer = (*TLSNetServer)(nil)
+
+// NewTLSNetServer returns a new initialised NetServer or an error.
+func NewTLSNetServer(options ...Option) (*TLSNetServer, error) {
```

Comment:
I think having 'default' or 'tls' like this implementation here seriously limits extensibility in the future, since each NetServer can only do one thing. It would hard to support a server that serves TLS but also does something else.